Letterbook.Web/Pages/Profile.cshtml.cs

```diff
@@ -11,49 +12,70 @@ namespace Letterbook.Web.Pages;
 [Authorize(Policy = Constants.AuthzPolicy)]
 public class Profile : PageModel
 {
-	private readonly IProfileService _profiles;
+	private readonly IProfileService _profileSvc;
 	private readonly ILogger<Profile> _logger;
 	private readonly CoreOptions _options;
+	private IAuthzProfileService _profiles;
+	private Models.Profile _profile;
+	private Models.Profile? _self;
 
-	public string BareHandle { get; set; }
-	public string Handle { get; set; }
-	public string DisplayName { get; set; }
-	public HtmlString Description { get; set; }
-	public Models.CustomField[] CustomFields { get; set; }
+	/// The full conventional fediverse @user@domain username.
+	/// Ostensibly globally unique
+	public string Handle => $"@{BareHandle}@{_profile.Authority}";
 
-	private protected Models.Profile? Prof { get; set; }
+	/// Just the username, with no @'s
+	public string BareHandle => _profile.Handle;
 
-	public int GetFollowerCount() => Prof!.FollowersCollection.Count;
-	public int GetFollowingCount() => Prof!.FollowingCollection.Count;
-	public string GetId() => Prof!.GetId25();
+	/// The non-unique display name
+	public string DisplayName => _profile.DisplayName;
+
+	/// The profile bio
+	public HtmlString Description => new(_profile.Description);
+
+	/// Public extra key-value fields
+	public Models.CustomField[] CustomFields => _profile.CustomFields;
+
+	/// The total number of profiles following this one
+	public Task<int> FollowerCount => _profiles.FollowerCount(_profile);
+
+	/// The total number of profiles followed by this one
+	public Task<int> FollowingCount => _profiles.FollowingCount(_profile);
+
+	/// The internal unique ID for this profile (used a lot in our APIs)
+	public string Id => _profile.GetId25();
+
+	public string? SelfId => User.Claims.FirstOrDefault(c => c.Type == "activeProfile")?.Value;
+
+	/// Whether this profile follows the User's activeProfile
+	public bool FollowsYou => _self?.FollowersCollection.Any() ?? false;
+
+	/// Whether the User's activeProfile follows this one
+	public bool YouFollow => _self?.FollowingCollection.Any() ?? false;
```

Comment:
I should note that in this case the `_self` profile's follow collections will be populated with the FollowerRelations to the `_profile`. So we _can_ just enumerate them. This is because `IProfileService.FindProfiles(Id, Id)` will eagerly join the FollowerRelations between those two profiles, if there are any.